xrpl/asyncio/transaction/reliable_submission.py

```diff
@@ -113,3 +124,108 @@ async def send_reliable_submission(
     return await _wait_for_final_transaction_outcome(
         transaction_hash, client, prelim_result, 0
     )
+
+
+def _is_signed(transaction: Transaction) -> bool:
+    """
+    Checks if a transaction has been signed.
+
+    Args:
+        transaction: the transaction to check.
+
+    Returns:
+        Whether the transaction has been signed
+    """
+    return (
+        transaction.signing_pub_key is not None and len(transaction.signing_pub_key) > 0
+    ) or (transaction.txn_signature is not None and len(transaction.txn_signature) > 0)
+
+
+def _decode_tx_blob(tx_blob: str) -> Transaction:
+    """
+    Decodes a transaction blob.
+
+    Args:
+        tx_blob: the tx blob to decode.
+
+    Returns:
+        The formatted transaction.
+    """
+    return Transaction.from_xrpl(decode(tx_blob))
+
+
+async def _get_signed_tx(
+    transaction: Union[Transaction, str],
+    client: Client,
+    wallet: Optional[Wallet] = None,
+    check_fee: bool = True,
+    autofill: bool = True,
+) -> Transaction:
+    """
+    Initializes a signed transaction for a submit request.
+
+    Args:
+        transaction: the transaction (or transaction blob) to be submitted.
+        client: the network client with which to submit the transaction.
+        wallet: the wallet with which to sign the transaction (optional, only needed
+        if the transaction is not signed).
+        check_fee: an optional bolean indicating whether to check if the fee is
+            higher than the expected transaction type fee. Defaults to True.
+        autofill: an optional boolean indicating whether to autofill the
+            transaction. Defaults to True.
+
+    Returns:
+        The signed transaction.
+    """
+    if type(transaction) is str:
+        transaction = _decode_tx_blob(transaction)
+
+    transaction = cast(Transaction, transaction)
```

Comment:
I believe this cast is unnecessary if you replace the if with `if isinstance(transaction, str):` (which is also the more Pythonic way to check type)